src/plugins/kibana/public/settings/sections/indices/add_data_steps/pipeline_setup/services/ingest.js

```diff
@@ -0,0 +1,58 @@
+import uiModules from 'ui/modules';
+import _  from 'lodash';
+import angular from 'angular';
+import { keysToCamelCaseShallow, keysToSnakeCaseShallow } from '../../../../../../../common/lib/case_conversion';
+
+const app = uiModules.get('kibana');
+
+app.service('ingest', function ($http) {
+  return {
+    simulatePipeline: simulatePipeline
+  };
+
+  function packagePipeline(pipeline) {
+    const requiredFields = ['input', 'processors'];
+    const cleanedPipeline = _.pick(pipeline, requiredFields);
+    return keysToSnakeCaseShallow(cleanedPipeline);
+  }
+
+  function packageProcessor(processor) {
+    const uiFields = [
+      '$$hashKey',
+      'collapsed',
+      'description',
+      'error',
+      'getDescription',
+      'inputObject',
+      'outputObject',
+      'parent',
+      'setParent',
+      'title',
+      'updateDescription'
```

Comment:
In light of my comment about keeping view data encapsulated, I don't think this belongs in the service. The service should be totally UI agnostic. We should be passing it plain objects that can be processed in a generic manner.